src/Bot.hs

```diff
@@ -17,21 +17,26 @@ import           Network.HTTP.Types.Status (Status(..))
 import           Network.URI (unEscapeString)
 import           Network.Wai (Application)
 import           System.Environment (getEnv)
+import           Text.Shakespeare.Text
 import           Types
 import           Utils
 import           Web.Scotty.Trans (ScottyT, get, post, scottyOptsT, status)
-import           Text.Shakespeare.Text
 import qualified Data.Text.Lazy as L
 import qualified Network.HTTP.Base as U
 import qualified Web.Scotty as S
 
 runApp :: IO ()
 runApp = do
   c <- C.load [Required "app.cfg"]
+  s <- engine <$> C.lookup c "engine"
   o <- opts
-  scottyOptsT o (runIO c) app' where
-    runIO :: Config -> ConfigM a -> IO a
-    runIO c m = runReaderT (runConfigM m) c
+  scottyOptsT o (runIO c s) app' where
+    runIO :: Config -> SearchEngine -> ConfigM a -> IO a
+    runIO c s m = runReaderT (runConfigM m) $ AppConfig c s
+
+engine :: Maybe Text -> SearchEngine
+engine (Just "hayoo") = Hayoo
```

Comment:
This feels like the kind of thing you'd want `Configurator` checking at parse time, not read time. Is that now feasible?